packages/api/portfolioDrafts/funding/createFundingStep.ts

```diff
@@ -35,7 +36,9 @@ export interface ClinValidationError {
  * @returns a collection of clin validation errors
  */
 export function validateFundingStepClins(fs: FundingStep): Array<ClinValidationError> {
-  return fs.clins.map(validateClin).reduce((accumulator, validationErrors) => accumulator.concat(validationErrors), []);
+  const clins: Array<Clin> = [];
+  fs.task_orders.forEach((taskOrder) => taskOrder.clins.forEach((clin) => clins.push(clin)));
```

Comment:
Correct me if I am wrong, but I think that this would be equivalent to:

```typescript
fs.task_orders.flatMap((taskOrder) => taskOrder.clins);
```

Reply:
I have not used a `flatMap` previously, but that makes sense and it does make it a lot cleaner. 

I did not add that test for `numOfX`. I'll included it. Thanks for the feed back.

Reply:
Changes regarding the `flatMap` are done. Tests for `numOfX` and `portfolio_name` have been added.